tests/test_zeros_synth.py

```diff
@@ -0,0 +1,187 @@
+            p_zero = 1.0 / (1.0 + torch.exp(-px_dropout))
+            p_dropout_infered.append(p_zero.cpu().numpy())
+
+            l_train_batch = torch.zeros((sample_batch.size(0), sample_batch.size(1), 100),
+                                        device=sample_batch.device)
+
+            for n_mc_sim in range(100):
+                p = px_rate / (px_rate + px_dispersion)
+                r = px_dispersion
+                l_train = torch.distributions.Gamma(concentration=r, rate=(1 - p) / p).sample()
+                l_train = torch.clamp(l_train, max=1e18)
+                X = torch.distributions.Poisson(l_train).sample()
+                l_train_batch[:, :, n_mc_sim] = l_train
+                p_zero = 1.0 / (1.0 + torch.exp(-px_dropout))
+                random_prob = torch.rand_like(p_zero)
+                X[random_prob <= p_zero] = 0
+
+            l_train_batch = torch.mean(l_train_batch, dim=(-1))
+
+            bio_zero_prob_batch = torch.exp(-l_train_batch)
+            tech_zero_prob_batch = p_zero
+
+            bio_zero_p.append(bio_zero_prob_batch.cpu().numpy())
+            tech_zero_p.append(tech_zero_prob_batch.cpu().numpy())
+            latent_reps.append(z.cpu().numpy())
+            poisson_params.append(l_train_batch.cpu().numpy())
+
+    latent_reps = np.concatenate(latent_reps)
+    bio_zero_p = np.concatenate(bio_zero_p)
+    tech_zero_p = np.concatenate(tech_zero_p)
+    bio_zero_tech_no = bio_zero_p * (1.0 - tech_zero_p)
+    tech_zero_bio_no = (1.0 - bio_zero_p) * tech_zero_p
+
+    # Final Step: Checking predictions
+    # Dropout checks
+    p_dropout_infered_all = np.concatenate(p_dropout_infered)
+    p_dropout_gt = synth_data.p_dropout.squeeze()
+    vmin = 0.0
+    vmax = 2.0*p_dropout_gt.max()
+    fig, axes = plt.subplots(ncols=2, nrows=2, figsize=(10, 10))
+    sns.heatmap(p_dropout_infered_all, vmin=vmin, vmax=vmax, ax=axes[0, 1])
+    axes[0, 1].set_title('Dropout Rate Predicted')
+    sns.heatmap(p_dropout_gt, vmin=vmin, vmax=vmax, ax=axes[0, 0])
+    axes[0, 0].set_title('Dropout Rate GT')
+
+    # Poisson Params checks
+    poisson_params = np.concatenate(poisson_params)
+    vmin = min(poisson_params_gt.min(), poisson_params.min())
+    vmax = max(poisson_params_gt.max(), poisson_params.max())
+    sns.heatmap(poisson_params, vmin=vmin, vmax=vmax, ax=axes[1, 1])
+    axes[1, 1].set_title('Poisson Distribution Parameter Predicted')
+
+    sns.heatmap(poisson_params_gt,  vmin=vmin, vmax=vmax, ax=axes[1, 0])
+    axes[1, 0].set_title('Poisson Distribution Parameter GT')
+    plt.savefig(os.path.join(folder, 'params_comparison.png'))
+    plt.close()
+
+    # TODO: Decrease test tolerances
+    l1_poisson = np.abs(poisson_params - poisson_params_gt).mean()
+    print('Average Poisson L1 error: ', l1_poisson)
+    assert l1_poisson <= 0.75, \
+        'High Error on Poisson parameter inference'
+    l1_dropout = np.abs(p_dropout_infered_all - synth_data.p_dropout).mean()
```

Comment:
Only check these assertions if the user specified --model-fit

Reply:
Normally fixed by 76fc30c